### PR TITLE
G62

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/digitalMatter/g62/uplink.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    digital.digitalInput1 = (bytes[4] & 0x2) !== 0;
+    digital.digitalInput2 = (bytes[4] & 0x4) !== 0;
+    digital.digitalOutput = (bytes[4] & 0x8) !== 0;
+    gps.headingDeg = bytes[8] * 2;
+    gps.speedKmph = bytes[9];
+    lifecycle.voltage = parseFloat((bytes[10] * 0.02).toFixed(3));
+
+    if (port === 1) {
+      digital.extVoltage = parseFloat(
+        (0.001 * (bytes[11] + bytes[12] * 256)).toFixed(3),
```
Comment (mariusgiger) [inline types/digitalMatter/g62/uplink.js:68]: explain this calculation

File: types/digitalMatter/g62/uplink.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+function hexToBytes(hex) {
+  let bytes;
+  let c;
+  for (bytes = [], c = 0; c < hex.length; c += 2) {
```
Comment (mariusgiger) [inline types/digitalMatter/g62/uplink.js:4]: why is the initialization done in the for loop for bytes and c?

File: types/digitalMatter/g62/gps.schema.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "Equals true if the GPS fix is current"
+    },
+    "headingDeg": {
+      "title": "heading",
+      "unit": "°",
+      "type": "number",
+      "description": "The direction where the sensor is heading"
+    },
+    "speedKmph": {
+      "title": "speedKmph",
```
Comment (mariusgiger) [inline types/digitalMatter/g62/gps.schema.json:41]: title lowercase?

File: types/digitalMatter/g62/gps.schema.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      "description": "Type of movement this trip",
+      "enum": ["NO_TRIP", "IGNITION", "MOVEMENT", "RUN"]
+    },
+    "gpsFixCurrent": {
+      "title": "GPS fix current",
+      "type": "boolean",
+      "description": "Equals true if the GPS fix is current"
+    },
+    "headingDeg": {
+      "title": "heading",
```
Comment (mariusgiger) [inline types/digitalMatter/g62/gps.schema.json:35]: title lowercase?

**Review by mariusgiger — COMMENTED** [review]
A few improvement suggestions

File: types/digitalMatter/g62/uplink.js (hunk 1 above)
Comment (J2thatsme) [reply types/digitalMatter/g62/uplink.js:68]: I don't have an explanation in the documentation for this '^^

Comment (mariusgiger) [reply types/digitalMatter/g62/uplink.js:69]: ok

**Review by mariusgiger — APPROVED** [review]

**Review by mariusgiger — APPROVED** [review]

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**